iree/modules/strings/strings_module.cc

```diff
@@ -67,23 +49,137 @@ class StringsModuleState final {
   Status Initialize() { return OkStatus(); }
 
   // strings.print(%str)
-  Status Print(vm::ref<vmstring_t> str) {
+  Status Print(vm::ref<string_t> str) {
     fwrite(str->value.data, 1, str->value.size, stdout);
     fputc('\n', stdout);
     fflush(stdout);
     return OkStatus();
   }
 
   // strings.i32_to_string(%value) -> %str
-  StatusOr<vm::ref<vmstring_t>> I32ToString(int32_t value) {
-    vm::ref<vmstring_t> new_string;
+  StatusOr<vm::ref<string_t>> I32ToString(int32_t value) {
+    vm::ref<string_t> new_string;
     std::string str = std::to_string(value);
     RETURN_IF_ERROR(
-        FromApiStatus(vmstring_create(iree_make_cstring_view(str.c_str()),
-                                      allocator_, &new_string),
+        FromApiStatus(string_create(iree_make_cstring_view(str.c_str()),
+                                    allocator_, &new_string),
+                      IREE_LOC));
+    return new_string;
+  }
+
+  const iree_string_view_t* PrintTensorHelper(const iree_string_view_t* strs,
```

Comment:
can you rewrite this to produce a string. It seems that having this factoring is a *lot* nicer:

StringTensorToString : string_tensor -> string
PrintString : string -> ()

Reply:
Added StringTensorToString and moved PrintTensor.